lib/http_ex/backend/mock/expectation.ex

```diff
@@ -184,11 +185,27 @@ defmodule HTTPEx.Backend.Mock.Expectation do
 
     {min_calls, max_calls} =
       case type do
-        :stub -> {0, :infinity}
-        :reject -> {0, 0}
+        :stub ->
+          {0, :infinity}
+
+        :reject ->
+          {0, 0}
+
         :assert ->
           min = Keyword.get(opts, :min_calls, 1)
-          max = Keyword.get(opts, :max_calls, Keyword.get(opts, :calls, 1))
+
+          max =
+            cond do
+              Keyword.has_key?(opts, :max_calls) ->
+                Keyword.get(opts, :max_calls)
+
+              Keyword.has_key?(opts, :min_calls) ->
+                :infinity
+
+              true ->
+                1
+            end
```

Comment:
This isn't great, but you are doing this to go around dialyzer right? Would dialyzer accept if you do this in two calls?

You are now missing the backwards compatibility with `:calls` that was there before.

```elixir
Keyword.get(opts, :max_calls) || Keyword.get(opts, :calls, 1)
```

The max to `:infinity` if there is a `min_calls` after not having a `max_calls` feels a bit weird.